### PR TITLE
Enhancement: Switch to PSR-4 autoloading

### PR DESCRIPTION
**Pull request by localheinz**
This PR

* [x] switches to PSR-4 autoloading

Follows #164.

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
> flattens the directory structure

:-1:

**Comment by bighappyface** [comment]
@localheinz @GrahamCampbell I am concerned about this PR as well. While I am all for PSR-4, flattening the directories could be really problematic for forked variations with patches applied in lieu of the historically slow turnaround for PR merges on this repo. IMO, PSR-4 is not out, but we will need to spend more time on the structural impacts.

**Comment by localheinz** [comment]
@bighappyface @GrahamCampbell 

Rebased and push-forced, so switching to PSR-4 now, but not flattening the directory structure!

File: composer.json [hunk 1]
```diff
@@ -44,10 +44,10 @@
         "phpdocumentor/phpdocumentor": "~2"
     },
     "autoload": {
-        "psr-0": { "JsonSchema": "src/" }
+        "psr-4": { "JsonSchema\\": "src/JsonSchema" }
```
Comment (GrahamCampbell) [inline composer.json]: Should end in a `/` for the dirnames in both cases.

**Comment by bighappyface** [comment]
+1

@GrahamCampbell you good with this?

**Comment by GrahamCampbell** [comment]
:+1:

**Comment by bighappyface** [comment]
Derp! Just saw @GrahamCampbell responded on this and rolled v1.4.4 :disappointed: 

Will merge and account for this in 1.4.5 once another source code-related change is merged.

**Comment by localheinz** [comment]
@bighappyface 

:+1:

**Pull request merged**